addons/project_purchase_stock/models/purchase_order.py

```diff
@@ -0,0 +1,16 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    def _prepare_picking(self):
+        res = super()._prepare_picking()
+        if not self.project_id:
+            return res
+        return {
+            'project_id': self.project_id.id,
+            **res
```

Comment:
```suggestion
            'project_id': self.project_id.id,
            **res,
```

However, the `project_id` key could be erased by the parent method, is it expected? I guess currently it is not really the case but if it is not expected I would suggest to do that:
```suggestion
            **res,
            'project_id': self.project_id.id,
```